### PR TITLE
Add support for domain in aftermarket auction status

### PR DESCRIPTION
**Pull request by gius80**
## Changes proposed in this Pull Request

This PR add support for expired domains in aftermarket auction state, adding notice message and badge in site domains list and domain setting page.

You can apply this diff to your sandbox to mock a domain in auction state D75409-code (_aftermarket-auction-domain.com_).

auction-list

auction-detail

## Testing instructions

- Build this branch locally or open the live Calypso link
- Select a site and then open site domains list (`upgrades/domains`)
- Verify that the notice message and badge are correctly showed as in the screenshots above

### CODE REVIEW

File: client/lib/url/support.js [hunk 1]
```diff
@@ -18,6 +18,7 @@ export const DOMAINS = `${ root }/domains`;
 export const DOMAIN_CANCEL = `${ root }/domains/#cancel-a-domain-name`;
 export const DOMAIN_EXPIRATION = `${ root }/domains/domain-expiration/`;
 export const DOMAIN_EXPIRATION_REDEMPTION = `${ root }/domains/domain-expiration/#renewing-a-domain-in-the-redemption-period`;
+export const DOMAIN_EXPIRATION_AUCTION = `${ root }/domains/domain-expiration/#expiry-auction`;
```
Comment (leonardost) [inline client/lib/url/support.js:20]: Minor nitpick: This should be defined above `DOMAIN_EXPIRATION_REDEMPTION` to keep the alphabetical order 🤓

File: client/lib/domains/resolve-domain-status.tsx [hunk 2]
```diff
@@ -208,6 +209,37 @@ export function resolveDomainStatus(
...
+						'Your domain expired over 30 days ago and has been offered for sale at auction. If it is not sold you may be able to restore the domain to your account by paying a redemption fee starting on {{strong}}%(renewableUntil)s{{/strong}}. Until then, you will not be able to make any changes or transfer the domain. {{a}}Learn more{{/a}}',
+						{
+							components: {
+								strong: <strong />,
+								a: (
+									<a
+										href={ DOMAIN_EXPIRATION_AUCTION }
+										rel="noopener noreferrer"
+										target="_blank"
+										onClick={ ( e ) => e.stopPropagation() }
```
Comment (leonardost) [inline client/lib/domains/resolve-domain-status.tsx]: I think we don't need this `onClick` handler

**Review by leonardost — APPROVED** [review]
LGTM, tested and works as expected! 💯 
Left just some minor comments.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7155923

Thank you @gius80 for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7155923).

**Pull request merged**